core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -65,7 +65,12 @@ default boolean schemaExists(ConnectorSession session, String schemaName)
     }
 
     /**
-     * Returns a table handle for the specified table name, or null if the connector does not contain the table.
+     * Returns a table handle for the specified table name, or {@code null} if the connector does not contain the table.
+     * Implementation should return {@code null} when {@code tableName} refers to other relation type (e.g. a view, or
+     * materialized view).
+     *
+     * @throws TrinoException implementation can throw this exception when {@code tableName} refers to a table that
+     * cannot be queried.
      */
     @Nullable
```

Comment:
I think we should return `Optional` here. Just an old thought to hit the bell and listen where it rings.

Reply:
out of scope for this pr